app/app.js

```diff
@@ -180,20 +183,45 @@ angular.module('app', ['angular-md5'])
         return false;
       });
 
-      if (currentCollectionIndexInCollections > 0) {
-        vm.prevCollection = collections[currentCollectionIndexInCollections - 1];
-        vm.prevCollectionFirstComic = _.find(comics, ['id', vm.prevCollection.comicIds[0]]);
-      } else {
+      //suggested revision for Klaus to check
+      // if (currentCollectionIndexInCollections > 0) {
+      //   vm.prevCollection = collections[currentCollectionIndexInCollections - 1];
+      //   vm.prevCollectionFirstComic = _.find(comics, ['id', vm.prevCollection.comicIds[0]]);
+      // } else {       
+      //   vm.prevCollection = undefined;
+      //   vm.prevCollectionFirstComic = undefined;
+      // }
+      var prevCollectionIndexInCollections;
+      var nextCollectionIndexInCollections;
+      prevCollectionIndexInCollections = currentCollectionIndexInCollections;
+      nextCollectionIndexInCollections = currentCollectionIndexInCollections;
+      prevCollectionIndexInCollections --;
+      nextCollectionIndexInCollections ++;
+
+      
+
+      if(_.isEqual(collections[currentCollectionIndexInCollections], _.first(collections))){
         vm.prevCollection = undefined;
         vm.prevCollectionFirstComic = undefined;
+      } else  {
+        vm.prevCollection = collections[prevCollectionIndexInCollections];
+        vm.prevCollectionFirstComic = _.find(comics, ['id', _.first(vm.prevCollection.comicIds)]);
       }
 
-      if (vm.collections[currentCollectionIndexInCollections + 1]) {
-        vm.nextCollection = collections[currentCollectionIndexInCollections + 1];
-        vm.nextCollectionFirstComic = _.find(comics, ['id', vm.nextCollection.comicIds[0]]);
-      } else {
+      //  And also for Klaus to check
+      // if (vm.collections[currentCollectionIndexInCollections + 1]) {
+      //   vm.nextCollection = collections[currentCollectionIndexInCollections + 1];
+      //   vm.nextCollectionFirstComic = _.find(comics, ['id', vm.nextCollection.comicIds[0]]);
+      // } else {
+      //   vm.nextCollection = undefined;
+      //   vm.nextCollectionFirstComic = undefined;
+      // }
+      if(_.isEqual(collections[currentCollectionIndexInCollections], _.last(collections))){
```

Comment:
If this works, it's good logic. `isEqual` can be a bit funny though so it will need good testing

Reply:
Okay, will do